clang/test/CodeGenSYCL/debug-info-srcpos-kernel.cpp

```diff
@@ -29,11 +29,12 @@ int main() {
 // CHECK: call spir_func void {{.*}}6__init{{.*}} !dbg [[LINE_A0]]
 // CHECK: call spir_func void @_ZZ4mainENKUlvE_clEv{{.*}} !dbg [[LINE_B0:![0-9]+]]
 // CHECK: ret void, !dbg [[LINE_C0:![0-9]+]]
+// CHECK: !DIFile(filename: "{{.*}}debug-info-srcpos-kernel.cpp"{{.*}})
```

Comment:
Is that change in this test due to a default constructor of the sampler?
I don't know debug info well enough, but I see that we did not check DIFile metadata two times before the patch. Now we do. @bwyma , @stevemerr , could you please confirm that this test still makes sense?

Reply:
> @bwyma , ... could you please confirm that this test still makes sense?

There is a bug where two DIFile entries are being emitted for the same file. It is a pre-existing issue.
The correct DIFile used to be emitted immediately following the "use_kernel_for_test" DISubprogram so it was easy to identify. I suspect something with this patch caused this DIFile to be emitted earlier in the printed IR and the only way to distinguish between the two DIFile entries in this test is to check for both.

Reply:
Thanks for the input @bwyma . Yeah, that's exactly what happened, the DIFile got reordered and now there are two before DISubprogram. Could you please confirm that this is ok by giving an approval?